Support/I2D2 Check/Check_2018.do

```diff
@@ -251,6 +251,7 @@ di "Checks for head==4"
 di "Household Size"
 assert hhsize==. | (hhsize>hhsize13  & hhsize!=1 & hhsize!=.) if head==4
```

Comment:
@alexandraqn Was this line here already, did you create it. What is it doing? I cannot follow what it is highlighting.

Reply:
Hi @gronert-m , 
Thank you for your question. After checking the original I2D2 code , I can tell you that lines 251 and 252 were created by the previous consultants. In general. I did not modify the I2D2 original code unless there was a serious problem such a repeated logical test (marital== 3 repeated 2 times in the same line)). 

I understand this line as follows:
the code wants to assert that the var hhsize takes the value of missing or that the var hhsize is bigger to "hhsize13"  and different to 1 and different to missing for the case of the observation that take the value of 4 (head=parents). As to why they tested this, I presume that it  may be because they want to make sure that hhsize for households with parents is bigger to the count of households with household head, spouse and children (hhsize13). 

There was little documentation on the reason for that part of the code. Yet,  I would be happy to discuss this further if my reasoning does not make sense.

Reply:
Thanks @alexandraqn for the explanation. @Junying-T had flagged this line yesterday and we were debating what the reason for this could be and we could not think of one. Imagine a household with a head, a child of the head and the parent of the head. In that case hhsize13 would be 3 (max of values between 1 and 3 in that household). The household size is 3 three as well so it fails the `hhsize > hhsize13` test while being a perfectly normal household. I would move to ignore this. I think in most cases this will not raise any issues anyway, but if it serves no real purpose it is not worth keeping. What are your thoughts?

Reply:
Perhaps someone thought that the code should compare to some hhsize average. And sure if it does not make sense then perhaps we commented out, of course making sure there is some hhsize check.